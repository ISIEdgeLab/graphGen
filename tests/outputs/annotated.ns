# Autogenerated by nsGen.py - Sun Jan 14 02:44:19 2018 UTC
set ns [new Simulator]
source tb_compat.tcl

set magi_str "sudo python /proj/edgect/magi/current/magi_bootstrap.py -fp /proj/edgect/magi/current/"

set click_str "/proj/edgect/exp_scripts/startClickDPDK.sh "

set my_start "touch /tmp/my_start"

# Enclaves

# Enclave 1
set server1 [$ns node]
tb-set-hardware $server1 cli_server_type
set traf11 [$ns node]
tb-set-hardware $traf11 cli_server_type
set traf12 [$ns node]
tb-set-hardware $traf12 cli_server_type
set traf13 [$ns node]
tb-set-hardware $traf13 cli_server_type
set traf14 [$ns node]
tb-set-hardware $traf14 cli_server_type
set traf15 [$ns node]
tb-set-hardware $traf15 cli_server_type
set traf16 [$ns node]
tb-set-hardware $traf16 cli_server_type
set traf17 [$ns node]
tb-set-hardware $traf17 cli_server_type
set traf18 [$ns node]
tb-set-hardware $traf18 cli_server_type
set ct1 [$ns node]
tb-set-node-os $ct1 Ubuntu1404-64-STD
tb-set-hardware $ct1 ct_type
set crypto1 [$ns node]
tb-set-node-os ${crypto1} Ubuntu1404-64-STD
tb-set-hardware ${crypto1} crypto_type

# Enclave 2
set server2 [$ns node]
tb-set-hardware $server2 cli_server_type
set traf21 [$ns node]
tb-set-hardware $traf21 cli_server_type
set traf22 [$ns node]
tb-set-hardware $traf22 cli_server_type
set traf23 [$ns node]
tb-set-hardware $traf23 cli_server_type
set traf24 [$ns node]
tb-set-hardware $traf24 cli_server_type
set traf25 [$ns node]
tb-set-hardware $traf25 cli_server_type
set traf26 [$ns node]
tb-set-hardware $traf26 cli_server_type
set traf27 [$ns node]
tb-set-hardware $traf27 cli_server_type
set traf28 [$ns node]
tb-set-hardware $traf28 cli_server_type
set ct2 [$ns node]
tb-set-node-os $ct2 Ubuntu1404-64-STD
tb-set-hardware $ct2 ct_type
set crypto2 [$ns node]
tb-set-node-os ${crypto2} Ubuntu1404-64-STD
tb-set-hardware ${crypto2} crypto_type

# Enclave 3
set server3 [$ns node]
tb-set-hardware $server3 cli_server_type
set traf31 [$ns node]
tb-set-hardware $traf31 cli_server_type
set traf32 [$ns node]
tb-set-hardware $traf32 cli_server_type
set traf33 [$ns node]
tb-set-hardware $traf33 cli_server_type
set traf34 [$ns node]
tb-set-hardware $traf34 cli_server_type
set traf35 [$ns node]
tb-set-hardware $traf35 cli_server_type
set traf36 [$ns node]
tb-set-hardware $traf36 cli_server_type
set traf37 [$ns node]
tb-set-hardware $traf37 cli_server_type
set traf38 [$ns node]
tb-set-hardware $traf38 cli_server_type
set ct3 [$ns node]
tb-set-node-os $ct3 Ubuntu1404-64-STD
tb-set-hardware $ct3 ct_type
set crypto3 [$ns node]
tb-set-node-os ${crypto3} Ubuntu1404-64-STD
tb-set-hardware ${crypto3} crypto_type

set vrouter [$ns node]
tb-set-node-os $vrouter Ubuntu1604-STD
tb-set-hardware $vrouter click_type
set control [$ns node]
tb-set-node-os $control Ubuntu1404-64-STD
tb-set-hardware $control cli_server_type

# External Nodes

# Lans
set lan1 [$ns make-lan "ct1 server1 traf11 traf12 traf13 traf14 traf15 traf16 traf17 traf18" 1000Mb 0ms]
set lan2 [$ns make-lan "ct2 server2 traf21 traf22 traf23 traf24 traf25 traf26 traf27 traf28" 1000Mb 0ms]
set lan3 [$ns make-lan "ct3 server3 traf31 traf32 traf33 traf34 traf35 traf36 traf37 traf38" 1000Mb 0ms]

# Internal Links
set link1 [$ns duplex-link $ct1 $crypto1 1000Mb 0.0ms DropTail]
set link2 [$ns duplex-link $ct2 $crypto2 1000Mb 0.0ms DropTail]
set link3 [$ns duplex-link $ct3 $crypto3 1000Mb 0.0ms DropTail]

# Egress Links
set elink1 [$ns duplex-link $crypto1 $vrouter 1000Mb 0.0ms DropTail]
set elink2 [$ns duplex-link $crypto2 $vrouter 1000Mb 0.0ms DropTail]
set elink3 [$ns duplex-link $crypto3 $vrouter 1000Mb 0.0ms DropTail]

tb-set-vlink-emulation "vlan"
tb-set-multiplexed $elink1 1
tb-set-noshaping $elink1 1
tb-set-multiplexed $elink2 1
tb-set-noshaping $elink2 1
tb-set-multiplexed $elink3 1
tb-set-noshaping $elink3 1

# External Node Links

# IPS

# IPs for Enclave 1
tb-set-ip-lan $traf11 $lan1 10.1.1.1
tb-set-ip-lan $traf12 $lan1 10.1.1.2
tb-set-ip-lan $traf13 $lan1 10.1.1.3
tb-set-ip-lan $traf14 $lan1 10.1.1.4
tb-set-ip-lan $traf15 $lan1 10.1.1.5
tb-set-ip-lan $traf16 $lan1 10.1.1.6
tb-set-ip-lan $traf17 $lan1 10.1.1.7
tb-set-ip-lan $traf18 $lan1 10.1.1.8
tb-set-ip-lan $server1 $lan1 10.1.1.9
tb-set-ip-lan $ct1 $lan1 10.1.1.100
tb-set-ip-link $ct1 $link1 10.1.2.1
tb-set-ip-link $crypto1 $link1 10.1.2.2

# IPs for Enclave 2
tb-set-ip-lan $traf21 $lan2 10.2.1.1
tb-set-ip-lan $traf22 $lan2 10.2.1.2
tb-set-ip-lan $traf23 $lan2 10.2.1.3
tb-set-ip-lan $traf24 $lan2 10.2.1.4
tb-set-ip-lan $traf25 $lan2 10.2.1.5
tb-set-ip-lan $traf26 $lan2 10.2.1.6
tb-set-ip-lan $traf27 $lan2 10.2.1.7
tb-set-ip-lan $traf28 $lan2 10.2.1.8
tb-set-ip-lan $server2 $lan2 10.2.1.9
tb-set-ip-lan $ct2 $lan2 10.2.1.100
tb-set-ip-link $ct2 $link2 10.2.2.1
tb-set-ip-link $crypto2 $link2 10.2.2.2

# IPs for Enclave 3
tb-set-ip-lan $traf31 $lan3 10.3.1.1
tb-set-ip-lan $traf32 $lan3 10.3.1.2
tb-set-ip-lan $traf33 $lan3 10.3.1.3
tb-set-ip-lan $traf34 $lan3 10.3.1.4
tb-set-ip-lan $traf35 $lan3 10.3.1.5
tb-set-ip-lan $traf36 $lan3 10.3.1.6
tb-set-ip-lan $traf37 $lan3 10.3.1.7
tb-set-ip-lan $traf38 $lan3 10.3.1.8
tb-set-ip-lan $server3 $lan3 10.3.1.9
tb-set-ip-lan $ct3 $lan3 10.3.1.100
tb-set-ip-link $ct3 $link3 10.3.2.1
tb-set-ip-link $crypto3 $link3 10.3.2.2

# Egress link IPS
tb-set-ip-link $crypto1 $elink1 10.1.10.1
tb-set-ip-link $vrouter $elink1 10.1.10.2
tb-set-ip-link $crypto2 $elink2 10.2.10.1
tb-set-ip-link $vrouter $elink2 10.2.10.2
tb-set-ip-link $crypto3 $elink3 10.3.10.1
tb-set-ip-link $vrouter $elink3 10.3.10.2

# External Node Link IPS 

# Write Tee Nodes

# Container Partitioning
tb-add-node-attribute $traf11 containers:partition 1
tb-add-node-attribute $traf12 containers:partition 1
tb-add-node-attribute $traf13 containers:partition 1
tb-add-node-attribute $traf14 containers:partition 1
tb-add-node-attribute $traf15 containers:partition 1
tb-add-node-attribute $traf16 containers:partition 1
tb-add-node-attribute $traf17 containers:partition 1
tb-add-node-attribute $traf18 containers:partition 1
tb-add-node-attribute $server1 containers:partition 1
tb-add-node-attribute $traf21 containers:partition 2
tb-add-node-attribute $traf22 containers:partition 2
tb-add-node-attribute $traf23 containers:partition 2
tb-add-node-attribute $traf24 containers:partition 2
tb-add-node-attribute $traf25 containers:partition 2
tb-add-node-attribute $traf26 containers:partition 2
tb-add-node-attribute $traf27 containers:partition 2
tb-add-node-attribute $traf28 containers:partition 2
tb-add-node-attribute $server2 containers:partition 2
tb-add-node-attribute $traf31 containers:partition 3
tb-add-node-attribute $traf32 containers:partition 3
tb-add-node-attribute $traf33 containers:partition 3
tb-add-node-attribute $traf34 containers:partition 3
tb-add-node-attribute $traf35 containers:partition 3
tb-add-node-attribute $traf36 containers:partition 3
tb-add-node-attribute $traf37 containers:partition 3
tb-add-node-attribute $traf38 containers:partition 3
tb-add-node-attribute $server3 containers:partition 3

tb-add-node-attribute $ct1 containers:partition 4
tb-add-node-attribute $crypto1 containers:partition 5
tb-add-node-attribute $ct2 containers:partition 6
tb-add-node-attribute $crypto2 containers:partition 7
tb-add-node-attribute $ct3 containers:partition 8
tb-add-node-attribute $crypto3 containers:partition 9

tb-add-node-attribute $vrouter containers:partition 10
tb-add-node-attribute $control containers:partition 11

# Embed Physical Nodes
tb-add-node-attribute $ct1 containers:node_type embedded_pnode
tb-add-node-attribute $crypto1 containers:node_type embedded_pnode
tb-add-node-attribute $ct2 containers:node_type embedded_pnode
tb-add-node-attribute $crypto2 containers:node_type embedded_pnode
tb-add-node-attribute $ct3 containers:node_type embedded_pnode
tb-add-node-attribute $crypto3 containers:node_type embedded_pnode

tb-add-node-attribute $vrouter containers:node_type embedded_pnode
tb-add-node-attribute $control containers:node_type embedded_pnode

# Start Commands

tb-set-node-startcmd $traf11 "$my_start; $magi_str"
tb-set-node-startcmd $traf12 "$my_start; $magi_str"
tb-set-node-startcmd $traf13 "$my_start; $magi_str"
tb-set-node-startcmd $traf14 "$my_start; $magi_str"
tb-set-node-startcmd $traf15 "$my_start; $magi_str"
tb-set-node-startcmd $traf16 "$my_start; $magi_str"
tb-set-node-startcmd $traf17 "$my_start; $magi_str"
tb-set-node-startcmd $traf18 "$my_start; $magi_str"
tb-set-node-startcmd $server1 "$my_start; $magi_str"
tb-set-node-startcmd $ct1 "$my_start; $magi_str"
tb-set-node-startcmd $crypto1 "$my_start; $magi_str"

tb-set-node-startcmd $traf21 "$my_start; $magi_str"
tb-set-node-startcmd $traf22 "$my_start; $magi_str"
tb-set-node-startcmd $traf23 "$my_start; $magi_str"
tb-set-node-startcmd $traf24 "$my_start; $magi_str"
tb-set-node-startcmd $traf25 "$my_start; $magi_str"
tb-set-node-startcmd $traf26 "$my_start; $magi_str"
tb-set-node-startcmd $traf27 "$my_start; $magi_str"
tb-set-node-startcmd $traf28 "$my_start; $magi_str"
tb-set-node-startcmd $server2 "$my_start; $magi_str"
tb-set-node-startcmd $ct2 "$my_start; $magi_str"
tb-set-node-startcmd $crypto2 "$my_start; $magi_str"

tb-set-node-startcmd $traf31 "$my_start; $magi_str"
tb-set-node-startcmd $traf32 "$my_start; $magi_str"
tb-set-node-startcmd $traf33 "$my_start; $magi_str"
tb-set-node-startcmd $traf34 "$my_start; $magi_str"
tb-set-node-startcmd $traf35 "$my_start; $magi_str"
tb-set-node-startcmd $traf36 "$my_start; $magi_str"
tb-set-node-startcmd $traf37 "$my_start; $magi_str"
tb-set-node-startcmd $traf38 "$my_start; $magi_str"
tb-set-node-startcmd $server3 "$my_start; $magi_str"
tb-set-node-startcmd $ct3 "$my_start; $magi_str"
tb-set-node-startcmd $crypto3 "$my_start; $magi_str"

tb-set-node-startcmd $vrouter "$click_str; $my_start; $magi_str"
tb-set-node-startcmd $control "sudo python /proj/edgect/exp_scripts/fixHosts.py; $my_start; $magi_str"

# Epilogue
$ns rtproto Static
$ns run
