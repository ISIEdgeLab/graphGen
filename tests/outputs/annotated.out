// Auto generated Click Template using GraphGen/clickGen.py at Sun Jan 14 02:44:19 2018 UTC
// Command line: ../graphgen/graphGen.py -n ./outputs/annotated.ns -o ./outputs/annotated.out ./inputs/annotated.xnet 


// Routers
router1 :: RadixIPLookup(${if1_ip} 3,
                         ${if1_16} ${if1_gw} 0,
                         ${if2_16} 1,
                         ${if3_16} 2,
                         10.100.150.2 1,
                         10.100.150.3 2,
                         10.100.150.1 3);

router2 :: RadixIPLookup(${if2_ip} 3,
                         ${if1_16} 0,
                         ${if2_16} ${if2_gw} 1,
                         ${if3_16} 2,
                         10.100.150.3 2,
                         10.100.150.1 0,
                         10.100.150.2 3);

router3 :: RadixIPLookup(${if3_ip} 3,
                         ${if1_16} 0,
                         ${if2_16} 1,
                         ${if3_16} ${if3_gw} 2,
                         10.100.150.2 1,
                         10.100.150.1 0,
                         10.100.150.3 3);


// Packet Classifiers
c1 :: Classifier(12/0800, 12/0806 20/0001, 12/0806 20/0002, -);
c2 :: Classifier(12/0800, 12/0806 20/0001, 12/0806 20/0002, -);
c3 :: Classifier(12/0800, 12/0806 20/0001, 12/0806 20/0002, -);
chost :: Classifier(12/0800, 12/0806 20/0001, 12/0806 20/0002, -);

// VLAN Multiplexing
vlanmux :: VlanSwitch(VLAN ${vlan1}, VLAN ${vlan2}, VLAN ${vlan3});

// Packet Arrival
$DPDKArrival
vlanmux[0] -> c1;
vlanmux[1] -> c2;
vlanmux[2] -> c3;
FromHost(fake0) -> chost;

// Packet Departure
$DPDKDeparture
out1 :: VLANEncap(${vlan1}) -> ${out_if1};
out2 :: VLANEncap(${vlan2}) -> ${out_if2};
out3 :: VLANEncap(${vlan3}) -> ${out_if3};

// Handle ARP
arpt :: Tee(4);

c1[1] -> ar1 :: ARPResponder(${if1_ip} ${if1_eth}) -> out1;
arpq1 :: ARPQuerier(${if1_ip}, ${if1_eth}) -> out1;
c1[2] -> arpt;
arpt[0] -> [1]arpq1;

c2[1] -> ar2 :: ARPResponder(${if2_ip} ${if2_eth}) -> out2;
arpq2 :: ARPQuerier(${if2_ip}, ${if2_eth}) -> out2;
c2[2] -> arpt;
arpt[1] -> [1]arpq2;

c3[1] -> ar3 :: ARPResponder(${if3_ip} ${if3_eth}) -> out3;
arpq3 :: ARPQuerier(${if3_ip}, ${if3_eth}) -> out3;
c3[2] -> arpt;
arpt[2] -> [1]arpq3;

chost[1] -> c1;
chost[2] -> arpt;

// Link Traffic Shaping
link_1_3_queue :: ThreadSafeQueue(1000);
link_1_3_bw :: LinkUnqueue(0ms, 1Gbps);
link_1_3_loss :: RandomSample(DROP 0.0);
link_3_1_queue :: ThreadSafeQueue(1000);
link_3_1_bw :: LinkUnqueue(0ms, 1Gbps);
link_3_1_loss :: RandomSample(DROP 0.0);
link_1_2_queue :: ThreadSafeQueue(1000);
link_1_2_bw :: LinkUnqueue(0ms, 1Gbps);
link_1_2_loss :: RandomSample(DROP 0.0);
link_2_1_queue :: ThreadSafeQueue(1000);
link_2_1_bw :: LinkUnqueue(0ms, 1Gbps);
link_2_1_loss :: RandomSample(DROP 0.0);
link_3_2_queue :: ThreadSafeQueue(1000);
link_3_2_bw :: LinkUnqueue(2ms, 1Mbps);
link_3_2_loss :: RandomSample(DROP 0.0);
link_2_3_queue :: ThreadSafeQueue(1000);
link_2_3_bw :: LinkUnqueue(2ms, 1Mbps);
link_2_3_loss :: RandomSample(DROP 0.0);

// Decrement TTL and send time exceeded replies
r1ttl_3 :: DecIPTTL;
r1ttl_3[1] -> ICMPError(10.100.150.1, timeexceeded) -> router1;
r3ttl_1 :: DecIPTTL;
r3ttl_1[1] -> ICMPError(10.100.150.3, timeexceeded) -> router3;
r1ttl_2 :: DecIPTTL;
r1ttl_2[1] -> ICMPError(10.100.150.1, timeexceeded) -> router1;
r2ttl_1 :: DecIPTTL;
r2ttl_1[1] -> ICMPError(10.100.150.2, timeexceeded) -> router2;
r1ttl_out_e1 :: DecIPTTL;
r1ttl_out_e1[1] -> ICMPError(10.100.150.1, timeexceeded) -> router1;
r3ttl_2 :: DecIPTTL;
r3ttl_2[1] -> ICMPError(10.100.150.3, timeexceeded) -> router3;
r2ttl_3 :: DecIPTTL;
r2ttl_3[1] -> ICMPError(10.100.150.2, timeexceeded) -> router2;
r3ttl_out_e3 :: DecIPTTL;
r3ttl_out_e3[1] -> ICMPError(10.100.150.3, timeexceeded) -> router3;
r2ttl_out_e2 :: DecIPTTL;
r2ttl_out_e2[1] -> ICMPError(10.100.150.2, timeexceeded) -> router2;

// Links
router1[1] -> r1ttl_2 -> SetTimestamp(FIRST true) -> link_1_2_queue -> CoDel() -> link_1_2_loss -> link_1_2_bw -> router2
router1[2] -> r1ttl_3 -> SetTimestamp(FIRST true) -> link_1_3_queue -> CoDel() -> link_1_3_loss -> link_1_3_bw -> router3
router2[0] -> r2ttl_1 -> SetTimestamp(FIRST true) -> link_2_1_queue -> CoDel() -> link_2_1_loss -> link_2_1_bw -> router1
router2[2] -> r2ttl_3 -> SetTimestamp(FIRST true) -> link_2_3_queue -> CoDel() -> link_2_3_loss -> link_2_3_bw -> router3
router3[0] -> r3ttl_1 -> SetTimestamp(FIRST true) -> link_3_1_queue -> CoDel() -> link_3_1_loss -> link_3_1_bw -> router1
router3[1] -> r3ttl_2 -> SetTimestamp(FIRST true) -> link_3_2_queue -> CoDel() -> link_3_2_loss -> link_3_2_bw -> router2

 // Teed Inputs and Outputs

 // Input from Teed interfaces is discarded

// Output Chains

// Send IP Packets to Routers
c1[0] -> Strip(14) -> CheckIPHeader(0) -> router1;
c3[0] -> Strip(14) -> CheckIPHeader(0) -> router3;
c2[0] -> Strip(14) -> CheckIPHeader(0) -> router2;
chost[0] -> Strip(14) -> CheckIPHeader(0) -> router1;

// Send out packets to Interfaces
router1[0] -> r1ttl_out_e1 -> arpq1;
router2[1] -> r2ttl_out_e2 -> arpq2;
router3[2] -> r3ttl_out_e3 -> arpq3;

// Local Delivery
toh :: ToHost(fake0);

router1[3] -> EtherEncap(0x0800, 1:1:1:1:1:1, 2:2:2:2:2:2) -> toh;
router3[3] -> EtherEncap(0x0800, 1:1:1:1:1:1, 2:2:2:2:2:2) -> toh;
router2[3] -> EtherEncap(0x0800, 1:1:1:1:1:1, 2:2:2:2:2:2) -> toh;
arpt[3] -> toh;

// Unknown packets to their death
c1[3] -> Print("${if1} Non IP") -> Discard;
c2[3] -> Print("${if2} Non IP") -> Discard;
c3[3] -> Print("${if3} Non IP") -> Discard;
chost[3] -> Print("Host Non IP") -> Discard;
